modules/markup/external/external.go

```diff
@@ -56,10 +56,10 @@ func (p *Renderer) SanitizerRules() []setting.MarkupSanitizerRule {
 
 // SanitizerDisabled disabled sanitize if return true
 func (p *Renderer) SanitizerDisabled() bool {
-	return p.DisableSanitizer
+	return !p.UseIFrame && p.DisableSanitizer
```

Comment:
That sounds more wrong than previously...
I'm in favor of the previous version.
Otherwise we will at somepoint get hard to debug problems.

Reply:
Since 1.17 hasn't been release, how about introducing a setting option `RENDER_PAGE_MODE=default/no-sanitizer/iframe` instead? To replace both DISABLE_SANITIZER+USE_IFRAME.


`RENDER_PAGE_MODE=default`: default mode as 1.16, use sanitizer
`RENDER_PAGE_MODE=no-sanitizer`: do not use sanitizer, render in the page directly
`RENDER_PAGE_MODE=iframe`: render in the iframe without sanitizer

Reply:
I read the code again, `renderFile` is only called once in `renderCode`. So the `DisplayInIFrame` doesn't need to be a member of the `Render`.

`renderCode` could follow the config `RENDER_PAGE_MODE` , output the `<iframe>` if the iframe is used.

Reply:
I have some thoughts about the refactoring of MarkupRender system.

First, the Render interface should be splitted to InternalRender and ExternalRender, then no confusing `SanitizerDisabled/DisplayInIFrame` function for internal renders.

Then, in the `renderFile`, it could do as type-cast to check if the matched render is an external render, if yes, check if it will do iframe rendering, if it will do, then output the iframe.

If you do not mind, I could try to propose a PR (today, to make sure no delay for 1.17 milestone)

Reply:
reverted.